yt_dlp/extractor/gmanetwork.py

```diff
@@ -0,0 +1,83 @@
+from .common import InfoExtractor
+from .dailymotion import DailymotionIE
+from .youtube import YoutubeIE
+
+
+class GMANetworkVideoIE(InfoExtractor):
+    _VALID_URL = r'https?://www\.gmanetwork\.com/\w+/\w+/\w+/(?P<id>\d+)/(?P<display_id>[\w-]+)/video'
+    _TESTS = [{
```

Comment:
No test for dailymotion?

Reply:
Is it a way to only restrict particular IE ? i used the `allowed_extractor` param in test but it always return ```yt_dlp.utils.DownloadError: ERROR: No suitable extractor (Youtube) found for URL 28BqW0AXPe0```. I don't have any links that only have `Dailymotion` url

Reply:
Then how did you assume embeds are youtube/dailymotion?

Reply:
It's from js source and manual testing as they have said `dailymotion_file`

Reply:
> I don't have any links that only have Dailymotion url

Means you have a URL with both youtube and dailymotion? In that case, the extractor should return a playlist

Reply:
Right, but didn't that would be considered unexpected by user as the link from Youtube and Dailymotion point to same video ?

Reply:
Can you give an example?

Reply:
Sure, for example this link (https://www.gmanetwork.com/fullepisodes/home/running_man_philippines/168677/running-man-philippines-catch-the-thief-full-chapter-2/video?section=home), from api request (`https://www.gmanetwork.com/api/data/content/video/168677`, the response return like this:
```json
{
  "video_id": "168677",
  "title": "Running Man Philippines: Catch the Thief (FULL CHAPTER 2)",
  "video_file": "28BqW0AXPe0",
  "video_image": "0",
  "description": "<p>Pa-intense na nang pa-intense ang laban sa 'Running Man Philippines!' Magawa kaya ng Runners na mabawi ang golden R-necklace mula kay Boobay? Mahuli na kaya nila ang spy sa grupo?<br />\n#RunningManPH #RunningManPhilippinesFullEpisode</p>\n\n<p>Consists of Full Episode 3 to 5:<br />\nRunning Man PH Full Episode 3 - aired Sept. 10, 2022<br />\nRunning Man PH Full Episode 4 - aired Sept. 11, 2022<br />\nRunning Man PH Full Episode 5 - aired Sept. 17, 2022</p>",
  "twitter_share": "0",
  "facebook_share": "0",
  "url": "tv/running_man_philippines/168677/running-man-philippines-catch-the-thief-full-chapter-2/video",
  "keywords": "youlol,running_man_philippines,glaiza_de_castro,ruru_madrid,buboy_villar,mikael_daez,lexi_gonzales,angel_guardian,kokoy_de_santos,full_episode,gma_network,gma_comedy,gma_entertainment,rss,rssfeed",
  "type": "1",
  "type_id": "1924",
  "post_date": "September 19, 2022",
  "auto_hide_date": "2022-09-19 17:53:42",
  "auto_hide_date_flag": "0",
  "comments": "0",
  "counter": null,
  "status": "1",
  "insert_timestamp": "2022-09-19 17:54:20",
  "update_timestamp": "2022-09-21 11:40:46",
  "count": null,
  "image": "FULL-EP-CMS_-20220919175418.jpg",
  "alt_text": "Running Man Philippines",
  "auto_url_flag": "0",
  "poll_id": "0",
  "poll_location": "0",
  "tile_expiry_date": "1950-11-30 00:00:00",
  "tile_type": "0",
  "display_in": "4",
  "pinned": "0",
  "pinned_expiry_date": "0000-00-00 00:00:00",
  "celebrity_news": "",
  "flipboard": "0",
  "is_ph": "0",
  "is_mobile": "1",
  "share_image": "FULL-EP-CMS_-20220919175418.jpg",
  "is_instant_article": "0",
  "author": "",
  "video_filename": "",
  "dailymotion": "0",
  "dm_selected": "0",
  "dailymotion_file": "k68aqqwX34H9SryiVOf",
  "geo_exception": "0",
  "default_source": "2",
  "image_orientation": "0",
  "background_position": "top center",
  "percentage": "100",
  "ga_author": "",
  "publish_date": "November 30, -0001 00:00:00",
  "fulleps_poster_image": "",
  "site_id": "1",
  "id": "168677",
  "contributors": [],
  "seo_title": "Running Man Philippines: Catch the Thief (FULL CHAPTER 2)",
  "socmed_title": "Running Man Philippines: Catch the Thief (FULL CHAPTER 2)",
  "page_title": "Running Man Philippines: Catch the Thief (FULL CHAPTER 2) | Running Man Philippines | TV",
  "override_url": "tv/running_man_philippines/168677/running-man-philippines-catch-the-thief-full-chapter-2/video",
  "kapuso_points": "0"
}
```
using yt-dlp for youtube id:
```shell
(python_proj2)> python -m yt_dlp -v "https://www.youtube.com/watch?v=28BqW0AXPe0"       
[debug] Command-line config: ['-v', 'https://www.youtube.com/watch?v=28BqW0AXPe0']
[debug] Portable config "yt-dlp\yt-dlp.conf": ['--netrc-location', 'yt-dlp_netrc']
[debug] User config: []
[debug] System config: []
[debug] Encodings: locale cp1252, fs utf-8, pref cp1252, out utf-8, error utf-8, screen utf-8 
[debug] yt-dlp version 2023.01.02 [d83b0ad80] (source) 
[debug] Lazy loading extractors is disabled
[debug] Git HEAD: 46d1c8d24 
[debug] Python 3.11.1 (CPython AMD64 64bit) - Windows-10-10.0.22000-SP0 (OpenSSL 1.1.1q  5 Jul 2022) 
[debug] exe versions: ffmpeg 5.1-full_build-www.gyan.dev (setts), ffprobe 5.1-full_build-www.gyan.dev, phantomjs 2.1.1 
[debug] Optional libraries: Cryptodome-3.14.1, certifi-2022.05.18.1, mutagen-1.45.1, sqlite3-2.6.0, websockets-10.3 
[debug] Proxy map: {} 
[debug] Loaded 1755 extractors 
[youtube] Extracting URL: https://www.youtube.com/watch?v=28BqW0AXPe0 
[youtube] 28BqW0AXPe0: Downloading webpage 
[youtube] 28BqW0AXPe0: Downloading android player API JSON 
[debug] Sort order given by extractor: quality, res, fps, hdr:12, source, vcodec:vp9.2, channels, acodec, lang, proto 
[debug] Formats sorted by: hasvid, ie_pref, quality, res, fps, hdr:12(7), source, vcodec:vp9.2(10), channels, acodec, lang, proto, filesize, fs_approx, tbr, vbr, abr, asr, vext, aext, hasaud, id
[debug] Default format spec: bestvideo*+bestaudio/best 
[info] 28BqW0AXPe0: Downloading 1 format(s): 248+251 
[debug] Invoking http downloader on "https://rr1---sn-uxa3vhnxa-uphl.googlevideo.com/videoplayback?expire=1672844737&ei=YUG1Y_7MBsKsigbS3om4Ag&ip=<ip>id=o-ACIslbe4te46CCqT-N1DseLzSFqu77G5jzX2_6g8tTz6&itag=248&source=youtube&requiressl=yes&mh=Nd&mm=31%2C29&mn=sn-uxa3vhnxa-uphl%2Csn-npoeenez&ms=au%2Crdu&mv=m&mvi=1&pcm2cms=yes&pl=24&initcwndbps=237500&spc=zIddbI0PRrYkx0jWvU_ChUEjdZwQnow&vprv=1&svpuc=1&mime=video%2Fwebm&gir=yes&clen=1337121086&dur=5312.500&lmt=1668869661128499&mt=1672822859&fvip=1&keepalive=yes&fexp=24007246&c=ANDROID&txp=5535434&sparams=expire%2Cei%2Cip%2Cid%2Citag%2Csource%2Crequiressl%2Cspc%2Cvprv%2Csvpuc%2Cmime%2Cgir%2Cclen%2Cdur%2Clmt&sig=AOq0QJ8wRQIhAPDIt6RiCdbsucTjzrXN59Dv_OR2JaTIdOIh7Ri2Mhv2AiA9KadGaQyiRLjLYEwHVCzcDzBrIetEYFsQZMBBBxxv0A%3D%3D&lsparams=mh%2Cmm%2Cmn%2Cms%2Cmv%2Cmvi%2Cpcm2cms%2Cpl%2Cinitcwndbps&lsig=AG3C_xAwRQIhAJI9wzYD5BQ6LqOIpohqYP4XmkB7ymG6Owp1DM1CtXmhAiB-nV8AnXVXHDKXPhKRky7IT-7F9Bcb8BeHULWH5nwTGA%3D%3D"
[debug] File locking is not supported. Proceeding without locking 
[download] Destination: Running Man Philippines： Catch the Thief (FULL CHAPTER 2) [28BqW0AXPe0].f248.webm 
[download]   0.9% of    1.25GiB at    2.41MiB/s ETA 08:44
ERROR: Interrupted by user
```

yt-dlp log for dailymotion:
```shell
$ yt-dlp -v "https://www.dailymotion.com/video/k68aqqwX34H9SryiVOf"
[debug] Command-line config: ['-v', 'https://www.dailymotion.com/video/k68aqqwX34H9SryiVOf']
[debug] User config: []
[debug] System config: []
[debug] Encodings: locale cp1252, fs utf-8, pref cp1252, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version 2023.01.02 [d83b0ad] (win32_exe)
[debug] Python 3.8.10 (CPython AMD64 64bit) - Windows-10-10.0.22000-SP0 (OpenSSL 1.1.1k  25 Mar 2021)
[debug] exe versions: ffmpeg 5.1-full_build-www.gyan.dev (setts), ffprobe 5.1-full_build-www.gyan.dev, phantomjs 2.1.1
[debug] Optional libraries: Cryptodome-3.16.0, brotli-1.0.9, certifi-2022.12.07, mutagen-1.46.0, sqlite3-2.6.0, websockets-10.4
[debug] Proxy map: {}
[debug] Loaded 1754 extractors
[dailymotion] Extracting URL: https://www.dailymotion.com/video/k68aqqwX34H9SryiVOf
[dailymotion] Downloading Access Token
[dailymotion] k68aqqwX34H9SryiVOf: Downloading media JSON metadata
[dailymotion] k68aqqwX34H9SryiVOf: Downloading metadata JSON
[dailymotion] k68aqqwX34H9SryiVOf: Downloading m3u8 information
[debug] Formats sorted by: hasvid, ie_pref, lang, quality, res, fps, hdr:12(7), vcodec:vp9.2(10), channels, acodec, filesize, fs_approx, tbr, vbr, abr, asr, proto, vext, aext, hasaud, source, id
[debug] Default format spec: bestvideo*+bestaudio/best
[info] k68aqqwX34H9SryiVOf: Downloading 1 format(s): http-1080-1
[debug] Invoking http downloader on "https://proxy-33.sv4.dailymotion.com/sec(jZ16B8kLT6Yl5mIFpOtsdN6Zhuj8elbxrt-5oDhKHOnEY5DAciWlGMaA421Mlg5m9v_MKuDzVXTrmZWWLTqqug)/video/155/709/506907551_mp4_h264_aac_fhd.mp4"
[debug] File locking is not supported. Proceeding without locking
[download] Destination: Running Man Philippines： Catch the Thief (FULL CHAPTER 2) [k68aqqwX34H9SryiVOf].mp4
[download]   0.6% of    3.84GiB at    2.63MiB/s ETA 24:44
ERROR: Interrupted by user

```